File: test/Dss-C/source/testConversion.c
//#include <io.h>
#include <stdio.h>
#include <string.h>

#include "heclib.h"
#include "TestDssC.h"



int testConversion()
{
	int DEBUG=0;
	int status;
	int number;
	char c[5];
	char filename_orig[200];
	char filename6[200];
	char filename6a[200];
	char filename7[200];
	char filename7a[200];

	long long ifltab6[600];
	long long ifltab7[600];

	memset(ifltab6,0,600*sizeof( long long));
	memset(ifltab7,0,600*sizeof( long long));
	


	stringCopy(filename_orig, sizeof(filename_orig), "sample.dss", _TRUNCATE);
	stringCopy(filename6, sizeof(filename6), "sample6.dss", _TRUNCATE);
	stringCopy(filename6a, sizeof(filename6a), "sample6a.dss", _TRUNCATE);
	stringCopy(filename7, sizeof(filename7), "sample7.dss", _TRUNCATE);
	stringCopy(filename7a, sizeof(filename7a), "sample7a.dss", _TRUNCATE);

	status = zopen6(ifltab6, filename_orig);
	if (status) {
		printf("Error opening %s, status = %d!\n", filename_orig, status);
		return  status;
	}

	zinquireChar(ifltab6, "nrec", c, (int)sizeof(c), &number);
	if (number <= 0) {
		printf("\n\nFirst file for conversion test is not correct.\n");
		printf("It should be %s\na version 6 file, copied from HEC-DSSVue/samples.\n\n", filename_orig);
		return -2;
	}

	remove(filename7);
	status = zopen7(ifltab7, filename7);	
	if (status) {
		printf("Error opening %s, status = %d!\n", filename7, status);
		return  status;
	}
	//zset("mlvl", " ", 5);

	status  = zcopyFile(ifltab6, ifltab7, 0);
    if (status) {
		printf("Error during conversion from 6 to 7, %s, status = %d!\n", filename7, status);
		return  status;
	}
    
	zclose(ifltab6);

	remove(filename6a);
	status = zopen6(ifltab6, filename6a);	
	if (status) {
		printf("Error opening %s, status = %d!\n", filename6a, status);
		return  status;
	}

	status  = zcopyFile(ifltab7, ifltab6, 0);
    if (status) {
		printf("Error during conversion from 7 to 6, %s, status = %d!\n", filename6a, status);
		return  status;
	}
    
	zclose(ifltab6);
	zclose(ifltab7);

	remove(filename7a);

	status = zconvertVersion(filename6a, filename7a);
	if (status) {
		printf("Error during conversion from 6 to 7, Loc 2, status = %d!\n", status);
		return  status;
	}

	status = zconvertVersion("sample7a.dss", "sample6a.dss");
	if (status) {
		printf("Error during conversion from 6 to 7, Loc 3, status = %d!\n", status);
		return  status;
	}
		
	zset("mlvl", " ", 4);

	printf("File conversion from 6 to 7 and 7 to 6 passed.\n");

	return 0;
}

File: test/Fortran/source/test_conversion.f90
     subroutine testConversion (messageUnit, status)
!
!
!      USE IFPORT
!      USE IFCORE
!
      implicit none
!
      
      integer messageUnit, status, ierr
      integer highFunction, lowFunction, errorCode, istatus, severity
       character c*5

      common /lchk/ lcheck
      logical lcheck
!
!
      integer nrec
      integer ifltabSize
      integer(8) ifltab1(600), ifltab2(600)        
      ifltabSize = 600   
!
!
     write(messageUnit, 10)
 10  format('*** Enter testConversion ***')
     lcheck = .true.
 !   Set the output message level to moderate
    ! call zset('mlvl', ' ', 4)
   !  call zopen(ifltab1, 'C:/temp/sample6.dss', status)
   !  ierr = DELFILESQQ ('C:/temp/sample7convert.dss')
   !  call zopen7(ifltab2, 'C:/temp/sample7convert.dss', status)

     !call zset('mlvl', ' ', 2)
     call zgetfileversion('sample.dss', status)
     if (status.ne.6) then
            write(messageUnit,*)' '
            write(messageUnit,*)'sample.dss is required for conversion test '
            write(messageUnit,*)'It must be the sample.dss file from DSSVue '
            write(messageUnit,*)'And must be a version 6 file'
            if (status.eq.0) then
                write(messageUnit,*)'That file does not exist'
            else if (status.eq.7) then
                write(messageUnit,*)'That file is a version 7 file'
            endif
            write(messageUnit,*)' '
            return 
     endif
     
     call zopen6(ifltab1, 'sample.dss', status)
     if (status.ne.0) return 
     c = ' '
     call zinqir(ifltab1, 'NREC', c, nrec)
     if (nrec.eq.0) then
        write(messageUnit,*)' '
        write(messageUnit,*)' First file for converion test is not correct.'
        write(messageUnit,*)' It should be the sample dss 6 file.'
        status = -1
        return
     endif
     call cdelet ('sample7.dss',ierr)
     call zopen7(ifltab2, 'sample7.dss', status)
     call zcopyfile(ifltab1, ifltab2, status)     
     if (status.eq.123) then
        call zclose(ifltab1)
        call CDELET ('sample6convert.dss',ierr)
        call zopen6(ifltab1, 'sample6convert.dss', status)     
        call zcopyfile(ifltab2, ifltab1, status)
     endif

!
!  
 800    continue
        if (status .ne. 0) go to 900
        call zclose(ifltab1)
        call zclose(ifltab2)
        Write(messageUnit,*)'test conversion  passed successfully!'
  
        return
!
!       Error return
 900    Continue
        Write(messageUnit, 820) status
        !Write(messageUnit, 811) filename1
        !Write(messageUnit, 811) filename2
 811    Format( 'File tested: ', a) 
 820    Format ('test conversion FAILED; error: ', I8)
        if (status .gt. 1000) then
            call zerrordecode(status, highFunction, lowFunction, errorCode, istatus, severity)
            write (*,*) 'highFunction   ',highFunction
            write (*,*) 'lowFunction   ',lowFunction
            write (*,*) 'errorCode   ',errorCode
            write (*,*) 'istatus   ',istatus
            write (*,*) 'severity   ',severity
        endif
      return
      end
